### PR TITLE
stop looking after match in import variables

### PR DESCRIPTION
**Pull request by jocgir**
In order to support multiple folders lookup when importing variables and stop as soon as we find a match, I changed the way globFiles works to accept an additional parameter specifying if the search should stop or continue after a match.

It is now possible to specify multiple source folders on `import_variables.source`. So `source` can be either a string or a list.

It is also possible to specify multiple nested under objects on `import_variables.nested_under`. `nested_under`  be either a string or a list. This is useful to avoid loading the same file multiple times.

Fixed concurrency problem with current logging until we change the logging library for something better. Also fixed a concurrency problem with the configuration files cache.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -67,7 +75,7 @@ github.com/bgentry/speakeasy v0.1.0/go.mod h1:+zsyZBPWlz7T6j88CTgSN5bM796AkVf0kB
 github.com/blang/semver v0.0.0-20170202183821-4a1e882c79dc/go.mod h1:kRBLl5iJ+tD4TcOOxsy/0fnwebNt5EWlYSAyrTnjyyk=
 github.com/blang/semver v0.0.0-20180723201105-3c1074078d32 h1:JN7dmcVkamyKIPCDJAtmfiBcpJCMyoLkxsFbxJs1r74=
 github.com/blang/semver v0.0.0-20180723201105-3c1074078d32/go.mod h1:kRBLl5iJ+tD4TcOOxsy/0fnwebNt5EWlYSAyrTnjyyk=
-github.com/bmatcuk/doublestar v1.1.5/go.mod h1:wiQtGV+rzVYxB7WIlirSN++5HPtPlXEo9MEoZQC/PmE=
```
[the remark this answers is not in the document]
Comment (jocgir) [reply go.sum]: In fact, I don't know why, but it looks like that the go version used to build alter this checksum. I have 1.13.1 and Denis too and we get a checksum error with the other go.sum.

I just changed the travis configuration to 1.13.x and it works now.

File: util/logger.go [hunk 2]
```diff
@@ -20,16 +21,22 @@ func CreateLogger(prefix string) *logging.Logger {
 
 // SetLoggingLevel sets the logging level
 func SetLoggingLevel(level int) {
+	loggingMutex.Lock()
```
[the remark this answers is not in the document]
Comment (jocgir) [reply util/logger.go]: Ok, good to know. Changing the logging library will be our best bet.

File: config/extension_import_variables.go [hunk 3]
```diff
@@ -20,15 +21,15 @@ import (
...
-	OptionalVarFiles []string          `hcl:"optional_var_files"`
-	NestedUnder      string            `hcl:"nested_under"`
-	TFVariablesFile  string            `hcl:"output_variables_file"`
-	FlattenLevels    *int              `hcl:"flatten_levels"`
-	EnvVars          map[string]string `hcl:"env_vars"`
-	OnCommands       []string          `hcl:"on_commands"`
+	Vars              []string          `hcl:"vars"`
+	RequiredVarFiles  []string          `hcl:"required_var_files"`
+	OptionalVarFiles  []string          `hcl:"optional_var_files"`
+	SourcesType       interface{}       `hcl:"source"`
```
[the remark this answers is not in the document]
Comment (jocgir) [reply config/extension_import_variables.go]: No, I intend to use that feature later to read local plan outputs.

File: config/extension_import_variables.go [hunk 4]
```diff
@@ -20,15 +21,15 @@ import (
...
+	NestedObjectsType interface{}       `hcl:"nested_under"`
```
[the remark this answers is not in the document]
Comment (jocgir) [reply config/extension_import_variables.go]: Same here.

File: config/extension_import_variables.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (julienduchesne) [reply config/extension_import_variables.go]: No, deprecate the singular version. Make it a list only. No need to support both string and list of string

**Pull request merged**